modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -147,7 +151,23 @@ private ReturnDelegate createReturnDelegate() {
       return VoidReturnDelegate.INSTANCE;
     }
 
-    return StringUtils.isBlank(target) ? new ValueReturnDelegate(muleContext) : new TargetReturnDelegate(target, muleContext);
+    return !isTargetPresent() ? new ValueReturnDelegate(muleContext) : new TargetReturnDelegate(target, muleContext);
+  }
+
+  private boolean isTargetPresent() {
+    if (isBlank(target)) {
+      return false;
+    }
+
+    if (target.startsWith(FLOW_VARS)) {
```

Comment:
This validation runs when the return delegate is created, which happens in the `doInitialise` method.